metrics-operator/converter/slo_converter.go

```diff
@@ -0,0 +1,259 @@
+package converter
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type SLOConverter struct {
+}
+
+func NewSLOConverter() *SLOConverter {
+	return &SLOConverter{}
+}
+
+type SLO struct {
+	Objectives []*Objective `yaml:"objectives" json:"objectives"`
+	TotalScore Score        `yaml:"total_score" json:"total_score"`
+}
+
+type Score struct {
+	Pass    string `yaml:"pass" json:"pass"`
+	Warning string `yaml:"warning" json:"warning"`
+}
+
+type Objective struct {
+	Name    string     `yaml:"sli" json:"sli"`
+	KeySLI  bool       `yaml:"key_sli,omitempty" json:"key_sli,omitempty"`
+	Weight  int        `yaml:"weight,omitempty" json:"weight,omitempty"`
+	Warning []Criteria `yaml:"warning,omitempty" json:"warning,omitempty"`
+	Pass    []Criteria `yaml:"pass,omitempty" json:"pass,omitempty"`
+}
+
+type Criteria struct {
+	Operators []string `yaml:"criteria,omitempty" json:"criteria,omitempty"`
+}
+
+func (c *SLOConverter) Convert(fileContent []byte, analysisDef string, namespace string) (string, error) {
+	//check that provider and namespace is set
+	if analysisDef == "" || namespace == "" {
+		return "", fmt.Errorf("--definition and --slo-namespace needs to be set for conversion")
```

Comment:
Nitpick: this component should not be concerned about how those parameters are passed, i.e. it does not know that it's called by a CLI. Also, the message should start, with a lower case character, so I suggest the following:
```suggestion
		return "", fmt.Errorf("missing arguments: 'definition' and 'namespace' needs to be set for conversion")
```